Remove old (unnecessary) legacy client handling.

lhofhansl opened the pull request:
See also #7548

This removes legacy client handling code that actually never worked before - it would treat all clients as legacy clients!
The largest change is just whitespace.

lhofhansl commented:
Needs a small rebase after #7548 
Let's agree we want to do this, if so I'll rebase.

SmallJoker commented:
Also needs code style cleanups.

lhofhansl commented:
@SmallJoker you mean because I touched it? Fair enough. It's just code removal and whitespace changes.

nerzhul commented:
if you can fix it can be nice, but yeah nice job to have looked at this

nerzhul commented on src/network/connection.cpp:
```diff
@@ -657,56 +657,54 @@ void Channel::UpdateTimers(float dtime,bool legacy_peer)
-			float successfull_to_lost_ratio = 0.0;
-			bool done = false;
+		/* dynamic window size */
+		float successfull_to_lost_ratio = 0.0;
```
0.0f

nerzhul commented on src/network/connection.cpp:
```diff
@@ -657,56 +657,54 @@ void Channel::UpdateTimers(float dtime,bool legacy_peer)
+		bool done = false;
 
-			if (packets_successfull > 0) {
-				successfull_to_lost_ratio = packet_loss/packets_successfull;
```
successful

nerzhul commented on src/network/connection.cpp:
```diff
@@ -657,56 +657,54 @@ void Channel::UpdateTimers(float dtime,bool legacy_peer)
+		}
+		else if (packet_loss > 0)
+		{
+			window_size = MYMAX(
```
std::max

nerzhul commented on src/network/connection.cpp:
```diff
@@ -657,56 +657,54 @@ void Channel::UpdateTimers(float dtime,bool legacy_peer)
+				/* don't even think about increasing if we didn't even
+				 * use major parts of our window */
+				if (reasonable_amount_of_data_transmitted)
+					window_size = MYMIN(
```
std::min

lhofhansl commented:
Lemme fix the code style and post a new version.

lhofhansl commented:
Rebased and fixed all code style issues I could see.

SmallJoker commented on src/network/connection.cpp:
```diff
@@ -627,17 +627,17 @@ void Channel::UpdatePacketTooLateCounter()
 	packet_loss_counter += dtime;
 
-	if (packet_loss_counter > 1.0)
+	if (packet_loss_counter > 1.0f)
```
Wrong bracket placement

SmallJoker commented on src/network/connection.cpp:
```diff
@@ -646,67 +646,65 @@ void Channel::UpdateTimers(float dtime,bool legacy_peer)
+		if (packets_successful > 0) {
+			successful_to_lost_ratio = packet_loss/packets_successful;
+		}
+		else if (packet_loss > 0)
```
`} else if (...) {`

SmallJoker commented on src/network/connection.cpp:
```diff
@@ -646,67 +646,65 @@ void Channel::UpdateTimers(float dtime,bool legacy_peer)
+		}
+
+		if (!done)
+		{
```
`if (..) {`
Same for the entire function part below.

SmallJoker commented on src/network/connection.cpp:
```diff
@@ -903,6 +901,10 @@ void Peer::Drop()
 UDPPeer::UDPPeer(u16 a_id, Address a_address, Connection* connection) :
 	Peer(a_address,a_id,connection)
 {
+	for(unsigned int i=0; i< CHANNEL_COUNT; i++)
```
`for (size_t i = 0; i < CHANNEL_COUNT` or use the C++11 range-based "for" loop.
The brackets can be omitted, otherwise please correct the placement.

lhofhansl commented:
Ok... Fixed more code style issues. Let's please stop here, or have a separate PR. I'm a fan of targeted PRs, they can be tracked and reverted more easily when needed.

SmallJoker commented:
@lhofhansl Thanks for fixing these issues. In general, only the touched lines should be fixed, but your additional changes surely don't hurt anyone.
Which clients would needs this legacy code anyway? We bumped the minimal proto version, but I'd like to know where this was actually used in-game.

lhofhansl commented:
I'll admit I'm a bit confused about what is compatible with what.

lhofhansl commented:
ca8ec46843da054e656d2f63b23d0b1695c023da
